internal/serve/httphandler/forgot_password_handler.go

```diff
@@ -55,11 +56,38 @@ func (h ForgotPasswordHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 	var forgotPasswordRequest ForgotPasswordRequest
 	err = json.NewDecoder(r.Body).Decode(&forgotPasswordRequest)
 	if err != nil {
-		httperror.BadRequest("invalid request body", err, nil).Render(w)
+		httperror.BadRequest("Invalid request body", err, nil).Render(w)
 		return
 	}
 
-	if !h.ReCAPTCHADisabled {
+	// validate request
+	v := validators.NewValidator()
+
+	v.Check(forgotPasswordRequest.Email != "", "email", "email is required")
+
+	if v.HasErrors() {
+		httperror.BadRequest("Request invalid", err, v.Errors).Render(w)
+		return
+	}
+
+	user, err := h.AuthManager.GetUserByEmail(ctx, forgotPasswordRequest.Email)
+	if err != nil {
+		if errors.Is(err, auth.ErrUserNotFound) {
+			// If we don't find the user by email, we just return an ok response
+			// to prevent malicious client from searching accounts in the system
+			log.Ctx(ctx).Errorf("Email in request not found: %s", forgotPasswordRequest.Email)
```

Comment:
👋  Leaking PII into the logs is considered a security problem. If a bad actor get access to the logs, they'd be able to recover emails from known users, and use that to narrow down their targets.

Whenever we log something like this, we usually truncate it with the function `utils.TruncateString`.

Reply:
done!